dashboard/common/db_util.py

```diff
@@ -1,20 +1,41 @@
 # Some utility functions used by other classes in this project
 import logging
 from datetime import datetime
-from typing import Dict, List, TypedDict, Union
+from typing import Dict, List, Literal, TypedDict, Union
+from urllib.parse import quote_plus
 
-from dateutil.parser import parse
 import django
+from sqlalchemy import create_engine
+from sqlalchemy.engine import Engine
+from dateutil.parser import parse
 from django.conf import settings
+from django.contrib.auth.models import User as DjangoUser
 from django_cron.models import CronJobLog
 
 from dashboard.models import Course, User
 
-from django.contrib.auth.models import User as DjangoUser
-
 logger = logging.getLogger(__name__)
 
 
+class DBParams(TypedDict):
+    NAME: str
+    USER: str
+    PASSWORD: str
+    HOST: str
+    PORT: int
+
+
+def create_sqlalchemy_engine(db_params: DBParams, type: Literal['mysql', 'postgres']) -> Engine:
```

Comment:
It looks like the type parameter isn't really needed as ENGINE will always be available in DBParams if you added it to the DBParams class. There would be a value of.

`django.db.backends.postgresql `or `django.db.bacakends.mysql.`. So seems like you could just drop this parameter and use

`if 'mysql' in new_db_params['ENGINE']:`

But I guess it isn't a big deal either way.

Reply:
I made the requested change, see https://github.com/tl-its-umich-edu/my-learning-analytics/pull/1426/commits/d22125cc31bffa45a993edad8d95f52f4c79dab0